unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/SearchChannelsWindowController.cs

```diff
@@ -215,5 +229,7 @@ private void HandleLeaveChannel(string channelId)
             dataStore.channels.channelLeaveSource.Set(ChannelLeaveSource.Search);
             OnOpenChannelLeave?.Invoke(channelId);
         }
+
+        private void OnUserProfileUpdate(UserProfile profile) => view.SetCreateChannelButtonsActive(channelsFeatureFlagService.IsAllowedToCreateChannels());
```

Comment:
It feels weird to call it when the profile changes. It should be an encapsulated functionality of the `channelsFeatureFlagService`. Perhaps it would be better that you create and suscribe to `channelsFeatureFlagService.OnChanged` event instead. What do you think?

Reply:
Done in https://github.com/decentraland/unity-renderer/pull/3195/commits/362047ca4ebeb02331568004f7e0502dcd0fc61d.